File: optisched-master/scripts/replacer.js
const fs = require('fs');
const path = require('path');

function processFile(filepath) {
    try {
        let content = fs.readFileSync(filepath, 'utf-8');
        if (!content.includes('<TouchableOpacity')) return;

        if (!content.includes('AnimatedPressable')) {
            const parts = filepath.replace(/\\\\/g, '/').split('/');
            const srcIndex = parts.indexOf('src');

            let relPath = '../components/AnimatedPressable';
            if (srcIndex !== -1) {
                const depth = parts.length - srcIndex - 2;
                relPath = depth > 0 ? '../'.repeat(depth) + 'components/AnimatedPressable' : './components/AnimatedPressable';
            }

            const lines = content.split('\n');
            let lastImportIdx = -1;
            for (let i = 0; i < lines.length; i++) {
                if (lines[i].startsWith('import ')) {
                    lastImportIdx = i;
                }
            }
            if (lastImportIdx !== -1) {
                lines.splice(lastImportIdx + 1, 0, `import { AnimatedPressable } from '${relPath}';`);
                content = lines.join('\n');
            }
        }

        content = content.replace(/<TouchableOpacity/g, '<AnimatedPressable');
        content = content.replace(/<\/TouchableOpacity>/g, '</AnimatedPressable>');
        content = content.replace(/TouchableOpacity,/g, '');
        content = content.replace(/, TouchableOpacity/g, '');
        content = content.replace(/import { TouchableOpacity } from 'react-native';\r?\n/g, '');
        content = content.replace(/import {  } from 'react-native';\r?\n/g, '');

        fs.writeFileSync(filepath, content, 'utf-8');
        console.log('Updated:', filepath);
    } catch (e) {
        console.error('Error on', filepath, e);
    }
}

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        if (fs.statSync(filepath).isDirectory()) {
            walk(filepath);
        } else if (filepath.endsWith('.tsx') || filepath.endsWith('.ts')) {
            processFile(filepath);
        }
    }
}

const srcDir = path.join(__dirname, 'src');
walk(srcDir);
console.log('Done script');

File: scripts/replacer.py
import os

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()

        if '<TouchableOpacity' not in content:
            return

        if 'AnimatedPressable' not in content:
            # Add import correctly
            parts = filepath.replace('\\', '/').split('/')
            try:
                src_index = parts.index('src')
                depth = len(parts) - src_index - 2
                rel_path = '../' * depth + 'components/AnimatedPressable' if depth > 0 else './components/AnimatedPressable'
            except ValueError:
                rel_path = '../components/AnimatedPressable'

            lines = content.split('\n')
            last_import_idx = -1
            for i, line in enumerate(lines):
                if line.startswith('import '):
                    last_import_idx = i
            
            if last_import_idx != -1:
                lines.insert(last_import_idx + 1, f"import {{ AnimatedPressable }} from '{rel_path}';")
                content = '\n'.join(lines)

        content = content.replace('<TouchableOpacity', '<AnimatedPressable')
        content = content.replace('</TouchableOpacity>', '</AnimatedPressable>')
        content = content.replace('TouchableOpacity,', '')
        content = content.replace(', TouchableOpacity', '')
        content = content.replace("import { TouchableOpacity } from 'react-native';\n", "")
        content = content.replace("import {  } from 'react-native';\n", "")
        
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print("Updated:", filepath)
    except Exception as e:
        print(f"Error on {filepath}:", e)

src_dir = os.path.join(os.getcwd(), 'src')
for root, _, files in os.walk(src_dir):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            process_file(os.path.join(root, file))

print("Done")
